### PR TITLE
add initialization for clientId

### PR DESCRIPTION
**Pull request by sychen52**

### CODE REVIEW

**Comment by sychen52** [comment]
#3

File: eglot-grammarly.el [hunk 1]
```diff
@@ -48,10 +48,17 @@ Link: https://github.com/znck/grammarly"
 
 (defun eglot-grammarly--server-command ()
   "Generate startup command for Grammarly language server."
-  (list "@emacs-grammarly/grammarly-languageserver" "--stdio"))
+  (list 'eglot-grammarly-server "grammarly-languageserver" "--stdio"))
```
Comment (jcs090218) [inline eglot-grammarly.el:51]: Can you change this back to `@emacs-grammarly/grammarly-languageserver`? AFAIK, the author of grammarly-languageserver doesn't use other editors so it's mostly like going to be unstable.

File: eglot-grammarly.el [hunk 2]
```diff
@@ -48,10 +48,17 @@ Link: https://github.com/znck/grammarly"
...
 
 (add-to-list 'eglot-server-programs
              `(,eglot-grammarly-active-modes . ,(eglot-grammarly--server-command)))
 
+(defclass eglot-grammarly-server (eglot-lsp-server) ()
+  :documentation "A custom class for grammarly langserver.")
+
+(cl-defmethod eglot-initialization-options ((server eglot-grammarly-server))
+  "Passes through required grammarly initialization options"
+    (list :clientId "client_BaDkMgx4X19X9UxxYRCXZo"))
```
Comment (jcs090218) [inline eglot-grammarly.el]: Let's extract this to a variable like how `lsp-grammarly` did.

```
(defconst lsp-grammarly-client-id "client_BaDkMgx4X19X9UxxYRCXZo"
  "Client ID is required for language server's activation.")
```

But rename it to eglot.

**Review by jcs090218 — COMMENTED** [review]
Some nits, others LGTM.

File: eglot-grammarly.el (hunk 1 above)
Comment (sychen52) [reply eglot-grammarly.el:51]: A quick question about this: After I install the server using npm as mentioned in the readme, I cannot find a binary by running "@emacs-grammarly/grammarly-languageserver --stdio". I can run it with "grammarly-languageserver --stdio"
Did I miss anything?

Comment (jcs090218) [reply eglot-grammarly.el:51]: Oh wait, please ignore my previous comment. I thought the command was for installation. My bad!

`grammarly-languageserver` should be good. 😅

**Review by jcs090218 — APPROVED** [review]

**Pull request merged**